artifactory_cleanup/rules/delete.py

```diff
@@ -102,3 +103,23 @@ def filter(self, artifacts):
         repositories = utils.build_repositories(artifacts)
         folders = utils.get_empty_folders(repositories)
         return folders
+
+
+class DeleteByRegex(Rule):
+    """
+    Remove artifacts by regex pattern.
+    """
+
+    def __init__(self, regex_pattern):
+        self.regex_pattern = rf"{regex_pattern}"
+
+    def aql_add_filter(self, filters):
+        print(filters)
+        print(type(filters))
+        return filters
+
+    def filter(self, artifacts: ArtifactsList) -> ArtifactsList:
+        for  val in artifacts[:]:
```

Comment:
We can name it explicitly - it's `artifact`, not just some value

Reply:
in that case `artifact['name']` means something, but with `val['name']` it's not obviousl